### ISSUE TITLE
[Discussion] Investigate moving to Azure CosmosDB Move from AZ Table Storage to Azure CosmosDB 

### ISSUE THREAD

**Issue by damsleth** [Puzzlepart/did#506]
### TLDR ###
We should investigate using Azure Cosmos DB as storage platform

### Discussion ###
[The Azure Cosmos DB Table API is nearly identical to Azure Table Storage](https://docs.microsoft.com/en-us/azure/cosmos-db/table-support), but provides higher throughput and scalability.

The [azure-storage](https://www.npmjs.com/package/azure-storage) SDK that we're already using for Azure Table Storage can also be used with the Cosmos DB Table API ([source](https://docs.microsoft.com/en-us/azure/cosmos-db/table-support#developing-with-the-azure-cosmos-db-table-api)), using the same CRUD-operations.

Many of the bulk-operations we're doing today, e.g. importing and exporting TimeEntries from one Table Storage table to another (~26k entries, 23mb), are already appearing cumbersome and "too big".

Likewise is querying the entire dataset of 25k items, taking more than 11 seconds to retrieve.

## What this discussion aims to achieve:
* Consensus on whether we should continue using Azure Table storage, or transition to Cosmos DB
* Figure out the cost of using Cosmos DB vs. Table Storage
* Estimate the complexity of transitioning from Table Storage to Cosmos DB
* Benchmark the performance difference of the two database solutions

### Sources
[Introduction to Azure Cosmos DB: Table API](https://docs.microsoft.com/en-us/azure/cosmos-db/table-introduction)
[Scaling Azure Table Storage to maximise throughput](https://www.ben-morris.com/the-poor-mans-cassandra-scaling-azure-table-storage-to-maximise-throughput)
[Cosmos DB Table API FAQ](https://docs.microsoft.com/en-us/azure/cosmos-db/table-api-faq)
[Azure Cosmos DB Pricing](https://azure.microsoft.com/en-us/pricing/details/cosmos-db/)
[Pricing: Request Units in Azure Cosmos DB](https://docs.microsoft.com/en-us/azure/cosmos-db/request-units)
[Troy Hunt loves Azure Table Storage](https://www.troyhunt.com/working-with-154-million-records-on/)

### Related issues ###
#154 
#476
